### PR TITLE
Replace some header includes with forward declarations

### PR DESCRIPTION
**Pull request by amosbird**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Replace some header includes with forward declarations to speed up compilation.

### CODE REVIEW

File: src/Interpreters/DatabaseCatalog.h [hunk 1]
```diff
@@ -1,15 +1,14 @@
...
+#include <Databases/TablesDependencyGraph.h>
 #include <Interpreters/Context_fwd.h>
 #include <Interpreters/StorageID.h>
-#include <Databases/TablesDependencyGraph.h>
 #include <Parsers/IAST_fwd.h>
 #include <Storages/IStorage_fwd.h>
-#include "Common/NamePrompter.h"
+#include <Common/NamePrompter.h>
 #include <Common/SharedMutex.h>
-#include "Storages/IStorage.h"
```
Comment (amosbird) [inline src/Interpreters/DatabaseCatalog.h:11]: IStorage.h should not be included.

**Comment by robot-ch-test-poll** [comment]
*This is an automated comment for commit bc977bbf4f42f14976ee9899023036090e0bd7f9 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/61003/bc977bbf4f42f14976ee9899023036090e0bd7f9/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

**Review by antaljanosbenjamin — APPROVED** [review]

**Comment by antaljanosbenjamin** [comment]
There is [one build](https://pastila.nl/?003522a6/a875f4bd4e8ee4d1dff13b69163111f1#ehbGzjky1khN64v4R8tiOw==) failure that might be related, could you please check it out?

**Comment by antaljanosbenjamin** [comment]
The coverage build failed with LLVM got by running out of memory. Let me restart it.

**Comment by antaljanosbenjamin** [comment]
In our private CI the ASAN build also failed with LLVM running out of memory. Either our CI is broken, or this PR changed something that makes our special builds run out of memory.

**Comment by antaljanosbenjamin** [comment]
There OOM killed compilations in other PRs, so I think the issue is not necessary in this PR.

**Comment by antaljanosbenjamin** [comment]
The builds should be less memory hungry now, so hopefully the checks will succeed.

File: src/Functions/coverage.cpp [hunk 2]
```diff
@@ -6,6 +6,7 @@
 #include <Columns/ColumnVector.h>
 #include <Columns/ColumnsNumber.h>
 #include <Columns/ColumnConst.h>
+#include <Columns/ColumnsNumber.h>
```
Comment (antaljanosbenjamin) [inline src/Functions/coverage.cpp:9]: I think this is the result of merging master, probably I did something wrong. Will fix in another PR.

**Comment by antaljanosbenjamin** [comment]
Not all of the checks succeeded, but nothing new failed in my opinion. Will merge.

**Pull request merged**